src/python/interpret/ext/extension_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2019 Microsoft Corporation
+# Distributed under the MIT software license
+
+import logging
+import pkg_resources
+import re
+
+module_logger = logging.getLogger(__name__)
+
+blackbox_key = "interpret_ext_blackbox"
+
+
+def _validate_class_name(proposed_class_name):
+    """
+    Used to validate class names before registration.
+
+    :param proposed_class_name: The string to name the class
+    :type proposed_class_name: str
+    """
+    # regex for class name came from
+    # https://stackoverflow.com/questions/10120295
+    # TODO: Support other languages
+    match = re.match(r"[a-zA-Z_][a-zA-Z_0-9]+", proposed_class_name)
+    if match is None or match.group(0) != proposed_class_name:
+        raise ValueError("Invalid class name {}. Class names must start with a "
+                         " letter or underscore. And can continue with letters, "
```

Comment:
nit: a little cleaner as "Class names must start with a letter or an underscore and can continue with letters, numbers, and underscores."